### DISCUSSION TITLE
CMYK colors

### DISCUSSION THREAD

**Discussion by stiebel** [ImageMagick/ImageMagick#7290] [Digital Image Processing]
Hi,
I have a very specific problem using CMYK colors.
I have to print large amount of images (several square meters a day) with a color laser printer.
I want to "autoclean" the printer by printing borders that "waste" toner.
So I want to print a border with the CMYK-color 0/100/0/100 (in percent)
But if I add this border to an existing CMYK image with
`convert image.tif -bordercolor 'CMYK(0,100,0,100)' -border 100x0 image.tif`
I get a border with only 0/0/0/100

Same problem with 
`convert image.tif -bordercolor 'CMYK(50,50,50,50)' -border '100x0' image.tif`
I get a border with 0/0/0/62

Using mpc instead of tif does not make a difference.

I understand, that for people a (0/0/0/62) looks like a (50/50/50/50) and is a clever conversion,
but I need the original numbers for my printer.

Is there a way to get what I want?

Roland

**Comment by snibgo** [comment]
I suggest using `magick` instead of `convert`, so IM will parse commands with v7 syntax instead of the very old v6 syntax. V7 syntax is more powerful and with better error-checking.

> So I want to print a border with the CMYK-color 0/100/0/100 (in percent)
-bordercolor 'CMYK(0,100,0,100)'

This won't give 100% magenta. Instead, it gives 100 on a scale of 0 to 255, which is about 39%. If you want 100%, then use either `255` or `100%`.

> But if I add this border to an existing CMYK image with
convert image.tif -bordercolor 'CMYK(0,100,0,100)' -border 100x0 image.tif
I get a border with only 0/0/0/100

How do you know the result border is 0/0/0/100? What software tells you this?

If I use IM to find the result color, I get the values as I specified them.

For example, I take an sRGB image, convert it to CMYK, and add a border.
```
magick toes.png -colorspace CMYK -bordercolor CMYK(0,100%,0,100%) -border 100x0 t2.tiff
```
What are the top-left 2x2 pixels?
```
magick t2.tiff -crop 2x2+0+0 txt:
# ImageMagick pixel enumeration: 2,2,0,65535,cmyk
0,0: (0,65535,0,65535)  #0000FFFF0000FFFF  cmyk(0,255,0,255)
1,0: (0,65535,0,65535)  #0000FFFF0000FFFF  cmyk(0,255,0,255)
0,1: (0,65535,0,65535)  #0000FFFF0000FFFF  cmyk(0,255,0,255)
1,1: (0,65535,0,65535)  #0000FFFF0000FFFF  cmyk(0,255,0,255)
```
As expected, magenta and black have the maximum values.

**Comment by stiebel** [comment]
Hi Snigbo,
thanks a lot for your help.
I followed your steps and indeed `converts` produces the wrong colors and` magick` produces the right colors.
I consider this as a bug, but there is no problem with it,
I will change my scripts from `convert` to `magick`.
Roland

**Comment by stiebel** [comment]
SOLVED
